File: java/designpattern/ObserverApp/src/com/techlabs/observer/test/TestAccount.java
package com.techlabs.observer.test;

import com.techlabs.observer.Account;
import com.techlabs.observer.EmailObserver;
import com.techlabs.observer.SmsObserver;

public class TestAccount {
	public static void main(String[] args) {
		Account account=new Account("Chandan", 50000.0);
		EmailObserver email=new EmailObserver();
		SmsObserver sms=new SmsObserver();
		
		account.register(email);
		account.register(sms);
		
		account.deposit(1000.0);
		account.deposit(2000.0);
		
		account.remove(sms);
		account.deposit(50000.0);
	}
}

File: dotNet/CSharp/EventApp/EventApp/Test/TestAccount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace EventApp.Test
{
    class TestAccount
    {
        public static void SendEmail(Account account)
        {
            Console.WriteLine("Email is Sent for {0} account", account);
        }

        public static void SendSMS(Account account)
        {
            Console.WriteLine("SMS is sent for {0} account", account);
        }

        static void Main(string[] args)
        {
            Account account = new Account("Chandan", 10000);
            account.OnBalanceChanged += SendEmail;
            account.OnBalanceChanged += SendSMS;

            account.Deposit(1000);
            account.Withdraw(500);
        }
    }
}
